azavea/echo-locator
# Feature/kak/map route interactions#27

## Overview

Display routes to neighborhoods on maps. Add map and sidebar interactions for setting the active neighborhood.


### Demo

![image](https://user-images.githubusercontent.com/960264/54693045-5e8ca380-4afc-11e9-8663-8e14473eaa3e.png)


### Notes

This displays non-active neighborhood routes also, but at a lower opacity. Otherwise, there is no indication which neighborhoods can be reached via transit within the time cutoff used by the static results (120 minutes).

The minimal neighborhood card details added here should be cleaned up with #73.

The data processing script got modified to write the zip code as the `id` property on the GeoJSON output.

The [FontAwesome icons](https://fontawesome.com/v4.7.0/icons/) available to the project have only one map marker icon (not the filled and unfilled in the wireframe), so I went with filled and unfilled circles instead. The green card header highlighting also was not in the wireframe, and we may want to remove or revisit that in a styling pass.


## Testing Instructions

 * Hovering over a neighborhood card header should set it active
 * Clicking a neighborhood centroid map marker should set the neighborhood as active
 * Updating form inputs should change routes as expected
 * Active neighborhood sidebar card should have green card header with filled circle icon
 * Route to active neighborhood should have start marker at origin and end marker on neighborhood


Closes #27
